client/my-sites/domains/paths.js

```diff
@@ -61,8 +61,19 @@ export function domainAddNew( siteName, searchTerm ) {
 	return path;
 }
 
-export function domainAddEmailUpsell( siteName, domainName ) {
-	return `/domains/add/${ domainName }/email/${ siteName }`;
+/**
+ * Builds a URL query string from an object. Handles null values.
+ *
+ * @param {Object} parameters - optional path prefix
+ * @returns {string} the corresponding query string
+ */
+const buildQueryString = ( parameters = {} ) =>
+	parameters ? stringify( parameters, { addQueryPrefix: true, skipNulls: true } ) : '';
```

Comment:
Since it's a general purpose function that could potentially be used by other functions from this file, I would move it at the very top of the latter as otherwise other developers will most likely miss it.